### PR TITLE
load big files in chunks 

### PR DESCRIPTION
**Pull request by mgckind**

### CODE REVIEW

File: easyaccess/eautils/fileio.py [hunk 1]
```diff
@@ -274,6 +274,21 @@ def read_file(filename):
...
+        dtypes = [df[c].dtype if df[c].dtype.kind != 'O'
+                  else np.dtype('S' + str(max(df[c].str.len())))
+                  for i, c in enumerate(df)]
+
+    if df.file_type == 'fits':
+        dtype = df[1].get_rec_dtype(vstorage='fixed')[0]
+        dtypes = [dtype[i] for i, d in enumerate(dtype.descr)]
+    return dtypes
+
+
```
Comment (kadrlica) [inline easyaccess/eautils/fileio.py:291]: Would it make sense for this function to live in `dtypes.py`?

File: easyaccess/easyaccess.py [hunk 2]
```diff
@@ -1845,31 +1849,59 @@ def insert_data(self, table, columns, values, dtypes=None):
...
+            load_args = load_parser.parse_args(line.split())
+        except SystemExit:
+            self.do_help('load_table')
+            return
+        if load_args.help:
+            self.do_help('load_table')
+            return
+        filename = self.get_filename(load_args.filename)
+        name = load_args.tablename
+        chunk = load_args.chunksize
```
Comment (kadrlica) [inline easyaccess/easyaccess.py:1896]: This is a cool use of `argparser`

File: easyaccess/easyaccess.py [hunk 3]
```diff
@@ -1956,23 +2068,65 @@ def do_append_table(self, line, name=''):
...
+                    total_rows += len(values)
+                except:
+                    break
+                try:
+                    if not done:
+                        self.insert_data(table, columns, values, dtypes, iteration)
+                        iteration += 1
+                except:
+                    print_exception()
+                    return
```
Comment (kadrlica) [inline easyaccess/easyaccess.py:2126]: Seems like a lot of code duplicated between `do_load_table` and `do_append_table`. Also, can't we abstract away the difference between `pandas` and `fits` data at this level? Specifically, providing a single interface to grabbing chunks (probably code to add in `fileio.py`)?

**Comment by kadrlica** [comment]
Is there a downside to loading by chunks? Should we always be doing it? What about a default `chunksize` set in the `ea_config.py` (maybe set to a fairly large value)?

File: easyaccess/eautils/fileio.py (hunk 1 above)
Comment (mgckind) [inline easyaccess/eautils/fileio.py:291]: Yes, although here is where those file_type are defined. Don't mind either way

File: easyaccess/easyaccess.py (hunk 2 above)
Comment (mgckind) [inline easyaccess/easyaccess.py:1896]: Yeah, and some other commands can be expanded as well to add optional arguments

File: easyaccess/easyaccess.py (hunk 3 above)
Comment (mgckind) [inline easyaccess/easyaccess.py:2126]: Yes, there is a lot. I wanted to get it done first but yes, there repetition inside load_table and append_table and among themselves. There is room to reduce these

**Comment by mgckind** [comment]
I didn't see any. if not chunksize is defined the default works as fast as the previous version.  Using a default chunksize might work as well but its a tricky number. Today I uploaded 25M 4-column file in one single shot. but 130M 12-column file in a 2-million row chunks. For fits you can get rows and columns before hand and make a guess, for csv is not straightforward. Also, it depends on memory resources as well. I'd bet 10M is a reasonable number to start with.

**Comment by kadrlica** [comment]
It seems like it would be better to have the `chunksize` constraint to be in in MB rather than rows (something like `upload_max_mb` similar to `outfile_max_mb`). Is there any memory safe way to get the number of rows in a csv file? If so, we could do something like `(upload_max_mb/filesize)*nrows = chunksize`.

**Comment by mgckind** [comment]
I'm not sure there is one very efficient without reading the file and even from the shell command wc -l is slow for very big files. You might get a guess by reading the first line to get the number of columns and possible data types and based on the size of the file estimate the number of rows or directly `chunksize`

**Comment by kadrlica** [comment]
I like that idea. What about reading the first ~100 (1000?) lines and guessing at the size per line. Then dividing the file into chunks based on the total file size? I really do think that chunking in units of MB rather than rows makes a lot more sense.

**Comment by kadrlica** [comment]
Any more thoughts about this? If I get time I could make a pass at writing the code...

**Pull request merged**